src/nvim/api/private/helpers.c

```diff
@@ -469,23 +472,33 @@ static inline void typval_encode_dict_end(EncodedData *const edata)
 #endif
 }
 
-#define TYPVAL_ENCODE_CONV_DICT_END() \
+#define TYPVAL_ENCODE_CONV_DICT_END(tv, dict) \
     typval_encode_dict_end(edata)
 
 #define TYPVAL_ENCODE_CONV_RECURSE(val, conv_type) \
     TYPVAL_ENCODE_CONV_NIL()
 
-// object_convert_one_value()
+#define TYPVAL_ENCODE_SCOPE static
+#define TYPVAL_ENCODE_NAME object
+#define TYPVAL_ENCODE_FIRST_ARG_TYPE EncodedData *const
+#define TYPVAL_ENCODE_FIRST_ARG_NAME edata
+// _object_convert_one_value()
 // encode_vim_to_object()
-TYPVAL_ENCODE_DEFINE_CONV_FUNCTIONS(static, object, EncodedData *const, edata)
+#include "nvim/eval/typval_encode.c.h"
```

Comment:
No, or the comment is needed for all occurrences. What is typval_encode.c.h file is “described” by `.c.h` extension: intended purpose of this extension is highlighting that this file is supposed to be serving essentially the same purpose as very lengthy macro (like the one used to exist in typval_encode.h).